org.omg.sysml/src/org/omg/sysml/util/repository/EMFModelRefresher.java

```diff
@@ -0,0 +1,257 @@
+		for (EStructuralFeature feature: getStructuralFeatuers(langElement))
+		{
+			if (!feature.isDerived() && remoteElement.containsKey(feature.getName())) {
+				if (feature instanceof EReference reference) {
+					transformStructuralFeature(langElement, remoteElement, reference, true, false);
+				}
+			}
+		}
+		
+		if (langElement instanceof FeatureTyping ft) {
+			ft.setTypedFeature(null);
+		}
+	}
+	
+	private void tranformAttributes(EObject langElement, Element remoteElement) {
+		for (EStructuralFeature feature: getStructuralFeatuers(langElement))
+		{
+			if (!feature.isDerived() && remoteElement.containsKey(feature.getName())) {
+				if (feature instanceof EAttribute reference) {
+					transformStructuralFeature(langElement, remoteElement, reference, false, false);
+				}
+			}
+		}
+	}
+	
+	private static Collection<EStructuralFeature> getStructuralFeatuers(EObject langElement){
+		EClass eClass = langElement.eClass();
+		
+		Set<EObject> allRedefined = eClass.getEAllStructuralFeatures().stream()
+				.map(f -> f.getEAnnotation("redefines")).filter(Objects::nonNull)
+				.flatMap(a -> a.getReferences().stream()).collect(Collectors.toSet());
+		
+		return eClass.getEAllStructuralFeatures().stream().filter(f -> !allRedefined.contains(f)).collect(Collectors.toList());
+	}
+	
+	@SuppressWarnings("unchecked")
+	private void transformStructuralFeature(EObject langElement,  Element remoteElement, EStructuralFeature feature, boolean isReference, boolean isContainment) {
+		if (!disabledStructuralFeatures.contains(feature.getName())) {
+			try {
+				if (feature.isMany()) {
+					var referenceList = (List<EObject>) langElement.eGet(feature, false);
+					Object value = remoteElement.get(feature.getName());
+					if (value instanceof Collection valueCollection) {
+						for (Object referenceValue: valueCollection) {
+							EObject referencedLangElement = resolveReference(referenceValue, isContainment);
+							if (referencedLangElement != null) {
+								referenceList.add(referencedLangElement);
+							}
+						}
+					} 
+				} else {
+					Object value = remoteElement.get(feature.getName());
+					if (value != null) {
+						langElement.eSet(feature, isReference? resolveReference(value, isContainment) : transformAttributeValue(value, feature));
+					}
+				}
+			}
+			catch (Exception e) {
+				System.out.printf("Unable to set structural feature %s::%s %n", feature.getEContainingClass().getName(), feature.getName());
+			}
+		}
+	}
+	
+	private EObject resolveReference(Object referenceValue, boolean isContainment) {
+		if (referenceValue instanceof Map referenceObjectMap) {
+			Object refUUID = referenceObjectMap.get("@id");
+			Element referencedElement = model.getElement(UUID.fromString(refUUID.toString()));
+			EObject referencedLangElement = isContainment? transform(referencedElement) : tracker.get(UUID.fromString(refUUID.toString()));
+			return referencedLangElement;
+		}
+		return null;		

```

Comment:
Can this be replaced by an Optional or some other structure? This design puts the burden on the caller to always check for null, but does not helps to ensure it is done.

Reply:
Handled in #616